### PR TITLE
Client Dev Inner Loop - Fake Services Example

### PR DESCRIPTION
**Pull request by saquib-rashid-msft**
# Pull Request Template

## What are you trying to address

* Building on PR #752 , taking one of the approaches detailed in that PR, Fake Services, and providing a detailed example of how to implement.

## Description of new changes

- Added new doc to /developer-experience

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [x] Label the pull request with the appropriate area(s)
- [x] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [x] Assign the project - Engineering Playbook Backlog
- [x] Assign the pull request to yourself

### CODE REVIEW

**Comment by saquib-rashid-msft** [comment]
@olasowemimo , @charleszipp , looking to have this PR reviewed. Please take a look when you can. As a note, this builds on #752 , the PR that Christoph just merged.

**Comment by olasowemimo** [comment]
Should we also add a reference from the client-dev-inner-loop.md to the fake-service-inner-loop.md. Other than that, looks good!

**Comment by saquib-rashid-msft** [comment]
Good point. I think I had it in there at one point but might have lost it when I merged main back in. I'll update and push out the change. Thanks @olasowemimo!

File: docs/developer-experience/client-app-inner-loop.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+## Stub / Fake Services
+
+Lower fidelity approaches run stub services, that could be generated
+from API specs, or run fake servers like JsonServer ([JsonServer.io: A
+fake json server API Service for prototyping and
+testing.](https://www.jsonserver.io/)) or Postman. All these services
+would respond with predetermined and configured JSON messages.
+
+An example of how to implement using Fake Services can be found [here](./recipes/fake-services-inner-loop.md).
```
Comment (saquib-rashid-msft) [inline docs/developer-experience/client-app-inner-loop.md:150]: @olasowemimo the reference to the fake services inner loop md is here.

**Review by olasowemimo — APPROVED** [review]

**Comment by charleszipp** [comment]
@saquib-rashid-msft this information would also be very useful from an automated testing perspective as well. You may consider a follow up PR to reference this from the testing section as well. This sets up the possibility of creating behavioral driven or consumer driven acceptance tests that do not depend on remote services which can also accelerate the feedback loop.

**Review by charleszipp — APPROVED** [review]

**Pull request merged**